sycl/include/CL/sycl/detail/plugin_impl.hpp

```diff
@@ -0,0 +1,75 @@
+//==--------------------- plugin_impl.hpp - SYCL platform-------------------==//
+//
+// Part of the LLVM Project, under the Apache License v2.0 with LLVM Exceptions.
+// See https://llvm.org/LICENSE.txt for license information.
+// SPDX-License-Identifier: Apache-2.0 WITH LLVM-exception
+//
+//===----------------------------------------------------------------------===//
+
+#pragma once
+#include <CL/sycl/detail/common.hpp>
+#include <CL/sycl/detail/pi.hpp>
+#include <CL/sycl/stl.hpp>
+
+__SYCL_INLINE namespace cl {
+namespace sycl {
+namespace detail {
+
+class plugin_impl {
```

Comment:
Why impl? Is there just "plugin" class?
Suggest renaming to something like PICaller.